### PR TITLE
Move different benchmark files into test/benchmark/ subdirectory

### PR DESCRIPTION
**Pull request by juj**
There already exists a `test/benchmark/` subdirectory, but some benchmarks were in the root `test/` directory.

### CODE REVIEW

File: test/test_benchmark.py [hunk 1]
```diff
@@ -871,98 +871,98 @@ def output_parser(output):
...
-                      shared_args=['-DBUILD_FOR_SHELL', '-I' + TEST_ROOT])
+                      shared_args=['-DBUILD_FOR_SHELL', '-I' + test_file('benchmark')])
 
   # Benchmarks the synthetic performance of calling function pointers.
   @non_core
   def test_native_function_pointers(self):
     def output_parser(output):
       return float(re.search(r'Total time: ([\d\.]+)', output).group(1))
-    self.do_benchmark('native_functions', read_file(test_file('benchmark_ffis.cpp')), 'Total time:',
+    self.do_benchmark('native_functions', read_file(test_file('benchmark', 'benchmark_ffis.cpp')), 'Total time:',
```
Comment (sbc100) [inline test/test_benchmark.py:885]: These days you can just do `test_file('benchmark/benchmark_ffis.cpp')`  (i.e. you can use forward slashes here.   python's pathlib will take care of making them backslashes on windows).  Here and elsewhere in this PR.

**Review by sbc100 — COMMENTED** [review]
Nice!

**Review by sbc100 — APPROVED** [review]

File: test/test_browser.py [hunk 2]
```diff
@@ -4355,11 +4355,11 @@ def test_wasm_locate_file(self):
 
   @also_with_threads
   def test_utf8_textdecoder(self):
-    self.btest_exit('benchmark_utf8.cpp', 0, args=['--embed-file', test_file('utf8_corpus.txt') + '@/utf8_corpus.txt', '-sEXPORTED_RUNTIME_METHODS=[UTF8ToString]'])
+    self.btest_exit('benchmark', 'benchmark_utf8.cpp', 0, args=['--embed-file', test_file('utf8_corpus.txt') + '@/utf8_corpus.txt', '-sEXPORTED_RUNTIME_METHODS=[UTF8ToString]'])
```
Comment (sbc100) [inline test/test_browser.py:4358]: Use `/` here and in test_core.py too.

**Review by sbc100 — APPROVED** [review]

File: test/test_browser.py (hunk 2 above)
Comment (juj) [reply test/test_browser.py:4358]: Ops, silly me, I was looking at a quick glance that there was only one place where that occurred.. will update.

**Pull request merged**